app/components/search-facet-daterange/component.js

```diff
@@ -76,16 +89,21 @@ export default Ember.Component.extend({
         }
     }),
 
-    buildQueryObject(start, end) {
-        let key = this.get('key');
-        return dateRangeFilter(key, start, end);
+    buildQueryObject(start, end, dateType) {
+        return dateRangeFilter(dateType, start, end);
     },
 
-    updateFilter(start, end) {
-        let key = this.get('key');
-        let value = start && end ? { start: moment(start).format(DATE_FORMAT), end: moment(end).format(DATE_FORMAT) } : { start: '', end: '' };
+    updateFilter(start, end, dateType) {
+        const options = this.get('filterOptions');
+        for (const option in options) {
+            if (options[option].filterBy === dateType) {
+                this.set('filterDisplay', options[option].display);
+                break;
+            }
+        }
+        let value = start && end ? { start: moment(start).format(DATE_FORMAT), end: moment(end).format(DATE_FORMAT), date: dateType } : { start: '', end: '', date: dateType };
```

Comment:
I have no strong feelings; I didn't even write that bit of code (at least for this PR)